nodes/modifier_make/subdivide_mk2.py

```diff
@@ -0,0 +1,399 @@
+        col.prop(self, "single_edge", toggle=True)
+
+        col = row.column(align=True)
+        col.prop(self, "only_quads", toggle=True)
+        col.prop(self, "smooth_even", toggle=True)
+
+    def draw_buttons(self, context, layout):
+        row = layout.row(align=True)
+        row.prop(self, 'sv_mute', icon="HIDE_" + ("ON" if self.sv_mute else "OFF"), icon_only=True)
+        row.prop(self, 'sv_frezze', icon="PAUSE", icon_only=True)
+
+        self.draw_common(context, layout)
+        if self.show_options:
+            self.draw_options(context, layout)
+
+    def draw_buttons_ext(self, context, layout):
+        self.draw_common(context, layout)
+        self.draw_options(context, layout)
+        layout.label(text="Ouput Numpy:")
+        r = layout.row()
+        for i in range(4):
+            r.prop(self, "out_np", index=i, text=socket_names[i], toggle=True)
+
+    def rclick_menu(self, context, layout):
+        '''right click sv_menu items'''
+        layout.prop_menu_enum(self, "falloff_type")
+        layout.prop_menu_enum(self, "corner_type")
+        layout.prop(self, "grid_fill", toggle=True)
+        layout.prop(self, "single_edge", toggle=True)
+
+
+        layout.prop(self, "only_quads", toggle=True)
+        layout.prop(self, "smooth_even", toggle=True)
+
+        layout.label(text="Ouput Numpy:")
+
+        for i in range(4):
+            layout.prop(self, "out_np", index=i, text=socket_names[i], toggle=True)
+    def sv_init(self, context):
+        inew = self.inputs.new
+        inew('SvVerticesSocket', "Vertices")
+        inew('SvStringsSocket', 'Edges')
+        inew('SvStringsSocket', 'Faces')
+        inew('SvStringsSocket', 'FaceData')
+        inew('SvStringsSocket', 'EdgeMask')
+
+        inew('SvStringsSocket', 'Cuts').prop_name = "cuts"
+        inew('SvStringsSocket', 'Smooth').prop_name = "smooth"
+        inew('SvStringsSocket', 'Fractal').prop_name = "fractal"
+        inew('SvStringsSocket', 'AlongNormal').prop_name = "along_normal"
+        inew('SvStringsSocket', 'Seed').prop_name = "seed"
+
+        onew = self.outputs.new
+        onew('SvVerticesSocket', 'Vertices')
+        onew('SvStringsSocket', 'Edges')
+        onew('SvStringsSocket', 'Faces')
+        onew('SvStringsSocket', 'FaceData')
+
+        onew('SvVerticesSocket', 'NewVertices')
+        onew('SvStringsSocket', 'NewEdges')
+        onew('SvStringsSocket', 'NewFaces')
+
+        onew('SvVerticesSocket', 'OldVertices')
+        onew('SvStringsSocket', 'OldEdges')
+        onew('SvStringsSocket', 'OldFaces')
+
+        self.update_mode(context)
+
+    def mute_node(self):
+        origin_socket = [0, 1, 2, 3, 0, 1, 2, 0, 1, 2]
+
+        for i, s in zip(origin_socket, self.outputs):
+            if self.inputs[i].is_linked:
+                s.sv_set(self.inputs[i].sv_get(deepcopy=False))
+
+    def freeze_node(self):
+        if 'storage' in self.mem.keys():
+            for m, s in zip(self.mem['storage'], self.outputs):
```

Comment:
i think you'll find self.mem['storage'] will be used for all instances of this node. so you make a second node and if it also uses `self.mem['storage']`, then both nodes will output the same data.

only when you do
```python
self.mem[hash(self)]['storage'] = ...
```
will each node have unique storage.

Reply:
so
```python
self.mem[hash(self)] = {}
self.mem[hash(self)]['storage'] = ...
```

Reply:
this is why nodes like the viewers and snlite implement `node_dict` in the seemingly convoluted way.

if your immediate thought is : " ok.. then why don't we have a nicer mechanism in svcommon for this " .
then i'm right there with ya.  i don't know.

Reply:
Another problem can be that the dictionary will be empty after rebooting Blender file. I don't know whether this was taken in account or not.

Reply:
If we would like to have Python attributes for Blender classes we should have function or class which could wright Python data into somewhere in Blender file and restore data from that written data. Also there is need in two triggers - open and save Blender file.

Reply:
when opening a blend `hash(self)` will indeed not be present in the node_dict ( `self.mem` ),  
1) either `self.mem` of all instances will be empty, or
2) self.mem isn't empty but the `hash(self)` of all nodes is different each time the blend is loaded

i'm not sure if 1 is true or 2.  i can't remember and can't test at the moment :).. i think it's `1` .

Reply:
It is 1. And we have `n_id` properties for nodes. It is robust `ID` of a node instance.

Reply:
yes `n_id` is preferable, see the implementation @vicdoval .

https://github.com/nortikin/sverchok/blob/7019c9bc5d78dd752debb8cd1319944abe905944/node_tree.py#L367

Reply:
rather than building that into svcommon, it will make more sense to add it to a new mixin, that nodes can inherit from purely if they need a convenient way to use a `node_dict`.

though, i'll argue that this Pullrequest is not the place to implement that mix-in

Reply:
The idea (for another PR) was it to be reloaded every time you opened the file.
We will see how is the best way to implement it...

Reply:
if a node is set to freeze, it will need to regenerate the data upon blend open and indeed iojson import.
- this can be done automated as a post-load operation, post-import
- can also be built into the node (using the mixin) by providing the function the produce the stored data, and if the node detects that it doesn't contain the data, then it will call that function and store it.

it's not super easy to explain, or indeed to implement in a foolproof way - but I think it is possible (as attested by the fact that SNLite implements it already...everyday :)  )